File: GFG/Maximum Sub-String after at most K changes/Maximum Sub-String after at most K changes.cpp
class Solution {
	public:
		int characterReplacement(string s, int k){
		    // Code here
		    
		      int ans = 0;
		  int l = 0;
		  vector<int>temp(26);
		  for(int i=0;i<s.size();i++){
		      temp[s[i]-'A']++;
		      int operation = i-l+1-*max_element(temp.begin(),temp.end());
		      if(operation>k){
		       temp[s[l]-'A']--;
		       l++;
		      }
		      else ans = max(ans,i-l+1);
		  }
		 return ans;
		}

};

File: GFG/Maximum Sub-String after at most K changes/Maximum Sub-String after at most K changes.java
class Solution
{
    public int characterReplacement(String s, int k)
    {
        // code here
        int[]arr=new int[26];
        int prev=0;
        int ans=0;
        for(int i=0;i<s.length();i++){
            arr[s.charAt(i)-'A']++;
            int curr=find(arr,k);
            if(curr==-1){
                arr[s.charAt(prev++)-'A']--;
            }else{
                ans=Math.max(ans,curr);
            }
        }
        return ans;
        
    }
    int find(int[]arr,int k){
        int same=0;
        int diff=0;
        for(int i:arr){
            if(i>same){
                diff+=same;
                same=i;
            }else{
                diff+=i;
            }
        }
        return diff<=k? same+diff : -1;
    }
}
